crates/chain/tests/test_tx_graph.rs

```diff
@@ -880,3 +881,57 @@ fn test_additions_last_seen_append() {
         );
     }
 }
+
+#[test]
+fn test_inserting_anchor_without_tx() {
+    let mut graph = TxGraph::<BlockId>::default();
+    let txs = [common::new_tx(0), common::new_tx(1)];
+    let chain: LocalChain = (0..=5)
+        .map(|ht| (ht, BlockHash::hash(format!("Block Hash {}", ht).as_bytes())))
+        .collect::<BTreeMap<u32, BlockHash>>()
+        .into();
+    let tip = chain.tip().unwrap();
+    let additions = graph.insert_anchor_preview(txs[0].txid(), chain.get_block(0).unwrap().clone());
+
+    // Additions should only include the Anchor for the Tx-Block 1 and 2
+    assert_eq!(
+        additions,
+        Additions {
+            tx: Default::default(),
+            txout: Default::default(),
+            anchors: BTreeSet::from([(chain.get_block(0).unwrap(), txs[0].txid()),]),
+            last_seen: Default::default(),
+        }
+    );
+
+    graph.apply_additions(additions);
```

Comment:
This is interesting. You are re-adding things that are already added. I don't think this should be here, but it would be something interesting to test in a separate test.

The result of adding things that are already added must be no changes being made.